contracts/integrations/test/exchange/transaction_protocol_fee_test.ts

```diff
@@ -0,0 +1,420 @@
+                    .getABIEncodedTransactionData();
+                nestedTransaction = await alice.signTransactionAsync({ data: recursiveData });
+            });
+
+            it('Alice batchExecuteTransactions => nested batchExecuteTransactions, Bob fillOrder, Charlie fillOrder; mixed protocol fees', async () => {
+                const transactions = [nestedTransaction, transactionB2, transactionC2];
+                const signatures = transactions.map(tx => tx.signature);
+                const txReceipt = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .awaitTransactionSuccessAsync({ from: alice.address, value: MIXED_FEES_WITH_REFUND });
+                expectedBalances.simulateFills(
+                    [order, order, order, order, order],
+                    [alice.address, bob.address, charlie.address, bob.address, charlie.address],
+                    txReceipt,
+                    deployment,
+                    MIXED_FEES_WITH_REFUND,
+                );
+            });
+            it('Alice batchExecuteTransactions => Bob fillOrder, nested batchExecuteTransactions, Charlie fillOrder; mixed protocol fees', async () => {
+                const transactions = [transactionB2, nestedTransaction, transactionC2];
+                const signatures = transactions.map(tx => tx.signature);
+                const txReceipt = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .awaitTransactionSuccessAsync({ from: alice.address, value: MIXED_FEES_WITH_REFUND });
+                expectedBalances.simulateFills(
+                    [order, order, order, order, order],
+                    [bob.address, alice.address, bob.address, charlie.address, charlie.address],
+                    txReceipt,
+                    deployment,
+                    MIXED_FEES_WITH_REFUND,
+                );
+            });
+            it('Alice batchExecuteTransactions => Bob fillOrder, Charlie fillOrder, nested batchExecuteTransactions; mixed protocol fees', async () => {
+                const transactions = [transactionB2, transactionC2, nestedTransaction];
+                const signatures = transactions.map(tx => tx.signature);
+                const txReceipt = await deployment.exchange
+                    .batchExecuteTransactions(transactions, signatures)
+                    .awaitTransactionSuccessAsync({ from: alice.address, value: MIXED_FEES_WITH_REFUND });
+                expectedBalances.simulateFills(
+                    [order, order, order, order, order],
+                    [bob.address, charlie.address, alice.address, bob.address, charlie.address],
+                    txReceipt,
+                    deployment,
+                    MIXED_FEES_WITH_REFUND,
+                );
+            });
+        });
+    });
+});
```

Comment:
I noticed there's no revert scenarios in this file. Are there any failure scenarios worth testing here? Maybe ensuring a revert if the protocol fee can't be paid or something?

Reply:
added 👍